### ISSUE TITLE
How to use resnet backbone? Is there some technique?

### ISSUE THREAD

**Issue by Yoline777** [facebookresearch/detectron2#1476]
```
_resnet_mapper = {18: resnet.resnet18, 50: resnet.resnet50, 101: resnet.resnet101}

class ResnetBackbone(Backbone):
    def __init__(self, cfg, input_shape=None, pretrained=True):
        super().__init__()
        depth = cfg.MODEL.RESNETS.DEPTH
        backbone = _resnet_mapper[depth](pretrained=pretrained)
        self.stage0 = nn.Sequential(backbone.conv1, backbone.bn1, backbone.relu, backbone.maxpool)
        self.stage1 = backbone.layer1
        self.stage2 = backbone.layer2
        self.stage3 = backbone.layer3
        self.stage4 = backbone.layer4

    def forward(self, x):
        x = self.stage0(x)
        x = self.stage1(x)
        x = self.stage2(x)
        x = self.stage3(x)
        x = self.stage4(x)
        return x

@BACKBONE_REGISTRY.register()
def build_torch_backbone(cfg, input_shape=None):
    """
    Build a backbone.

    Returns:
        an instance of :class:`Backbone`
    """
    if input_shape is None:
        input_shape = ShapeSpec(channels=len(cfg.MODEL.PIXEL_MEAN))

    backbone = ResnetBackbone(cfg, input_shape)
    assert isinstance(backbone, Backbone)
    return backbone
```
```
BACKBONE:
    NAME: "build_resnet_backbone"
  RESNETS:
    OUT_FEATURES: ["res5"]
```
When I use  resnet which from torchvision. The projects run with no error. But when I change to build_resnet_backbone from detectron2. The loss become NaN. This problem has troubled me for half a month. I want to know is there some tricks that detectron2 use. What I should set to make it run normally?

**Comment by ppwwyyxx** [comment]
Maybe because you use `pretrained=True` for torchvision.

As the [issue template](https://github.com/facebookresearch/detectron2/blob/master/.github/ISSUE_TEMPLATE/unexpected-problems-bugs.md) mentions:

> If you expect the model to converge / work better, note that we do not give suggestions on how to train a new model. Only in one of the two conditions we will help with it: (1) You're unable to reproduce the results in detectron2 model zoo. (2) It indicates a detectron2 bug.
